### PR TITLE
add support for new timestamp format

### PR DESCRIPTION
**Pull request by ritalwar**
- Enhancement

## What does this PR do?

This PR supports RabbitMQ new timestamp format with nanosecond precision and timezone "2022-03-23 13:16:58.000369+03:00".
Previous timestamp format "2019-04-12 10:11:15.094".
New DATETIME pattern is replaced by TIMESTAMP_ISO8601.

## Why is it important?

This change is required as latest rabbitmq version switched to new default timestamp format.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Related issues

- Closes #https://github.com/elastic/integrations/issues/2905
- Relates #https://github.com/elastic/integrations/issues/2905

### CODE REVIEW

File: x-pack/filebeat/module/rabbitmq/log/test/test.log [hunk 1]
```diff
@@ -1,79 +1,26 @@
...
-    xmerl
-    recon
-    lager
-    goldrush
-    compiler
-    syntax_tools
-    stdlib
-    kernel
-2019-04-12 10:01:01.031 [info] <0.8.0> Server startup complete; 6 plugins started.
- * rabbitmq_stomp
```
Comment (lalit-satapathy) [inline x-pack/filebeat/module/rabbitmq/log/test/test.log:62]: These types also.

File: x-pack/filebeat/module/rabbitmq/log/test/test.log-expected.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [
     {
-        "@timestamp": "2019-04-03T11:13:15.076-02:00",
+        "@timestamp": "2023-01-03T05:20:19.811-02:00",
```
Comment (lalit-satapathy) [inline x-pack/filebeat/module/rabbitmq/log/test/test.log-expected.json:3]: There is no difference in the @timestamp before vs after?

File: x-pack/filebeat/module/rabbitmq/log/test/test.log [hunk 3]
```diff
@@ -1,79 +1,26 @@
...
-2019-04-12 10:00:54.579 [info] <0.374.0> Stopping message store for directory '/usr/local/var/lib/rabbitmq/mnesia/rabbit@localhost/msg_stores/vhosts/628WB79CIFDYO9LJI6DKMI09L/msg_store_persistent'
-2019-04-12 10:00:54.588 [info] <0.374.0> Message store for directory '/usr/local/var/lib/rabbitmq/mnesia/rabbit@localhost/msg_stores/vhosts/628WB79CIFDYO9LJI6DKMI09L/msg_store_persistent' is stopped
-2019-04-12 10:00:54.589 [info] <0.371.0> Stopping message store for directory '/usr/local/var/lib/rabbitmq/mnesia/rabbit@localhost/msg_stores/vhosts/628WB79CIFDYO9LJI6DKMI09L/msg_store_transient'
-2019-04-12 10:00:54.598 [info] <0.371.0> Message store for directory '/usr/local/var/lib/rabbitmq/mnesia/rabbit@localhost/msg_stores/vhosts/628WB79CIFDYO9LJI6DKMI09L/msg_store_transient' is stopped
-2019-04-12 10:00:54.606 [info] <0.43.0> Application rabbit exited with reason: stopped
-2019-04-12 10:00:54.615 [info] <0.1398.0> Successfully stopped RabbitMQ and its dependencies
-2019-04-12 10:00:54.615 [info] <0.1398.0> Halting Erlang VM with the following applications:
```
[the remark this answers is not in the document]
Comment (ritalwar) [reply x-pack/filebeat/module/rabbitmq/log/test/test.log:44]: Updated

File: x-pack/filebeat/module/rabbitmq/log/test/test.log-expected.json (hunk 2 above)
Comment (ritalwar) [reply x-pack/filebeat/module/rabbitmq/log/test/test.log-expected.json]: The @timestamp field is parsed from timestamp field, the actual change is in timestamp field.

File: x-pack/filebeat/module/rabbitmq/log/test/test.log [hunk 4]
```diff
@@ -1,79 +1,169 @@
...
+2023-01-24 10:38:49.499616+00:00 [info] <0.229.0> Running boot step rabbit_management_load_definitions defined by app rabbitmq_management
+2023-01-24 10:38:49.499816+00:00 [info] <0.723.0> Resetting node maintenance status
+2023-01-24 10:38:49.519074+00:00 [info] <0.782.0> Management plugin: HTTP (non-TLS) listener started on port 15672
+2023-01-24 10:38:49.519174+00:00 [info] <0.810.0> Statistics database started.
+2023-01-24 10:38:49.519212+00:00 [info] <0.809.0> Starting worker pool 'management_worker_pool' with 3 processes in it
+2023-01-24 10:38:49.524893+00:00 [info] <0.824.0> Prometheus metrics: HTTP (non-TLS) listener started on port 15692
+2023-01-24 10:38:49.525012+00:00 [info] <0.723.0> Ready to start client connection listeners
+2023-01-24 10:38:49.525875+00:00 [info] <0.868.0> started TCP listener on [::]:5672
+ completed with 4 plugins.
+2023-01-24 10:38:49.664998+00:00 [info] <0.723.0> Server startup complete; 4 plugins started.
```
[the remark this answers is not in the document]
Comment (ritalwar) [reply x-pack/filebeat/module/rabbitmq/log/test/test.log:63]: updated.

**Pull request merged**